src/Swashbuckle.AspNetCore.Swagger/DependencyInjection/SwaggerBuilderExtensions.cs

```diff
@@ -25,5 +38,70 @@ public static class SwaggerBuilderExtensions
 
             return app;
         }
+#if NETCOREAPP3_0
+        public static IEndpointRouteBuilder MapSwagger(this IEndpointRouteBuilder endpoints, string pattern)
+        {
+            if (!RoutePatternFactory.Parse(pattern).Parameters.Any(x => x.Name == "documentName"))
+            {
+                throw new ArgumentException("Pattern must contains '{documentName}' parameter", nameof(pattern));
+            }
+
+            endpoints.MapGet(pattern, InvokeAsync);
+
+            return endpoints;
+        }
+
+        public static IEndpointRouteBuilder MapSwagger(this IEndpointRouteBuilder endpoints)
+        {
+            return endpoints.MapSwagger("/swagger/{documentName}/swagger.json");
+        }
+
+        private async static Task InvokeAsync(HttpContext context)
```

Comment:
You can reuse logic in the middleware and avoid duplication by doing this:

https://github.com/dotnet/aspnetcore/blob/658b37d2bd3b0da2c07e25b53fa5c97ce8d656d3/src/Middleware/HealthChecks/src/Builder/HealthCheckEndpointRouteBuilderExtensions.cs#L75-L77